processor/cumulativetodeltaprocessor/config.go

```diff
@@ -19,26 +19,50 @@ import (
 	"time"
 
 	"go.opentelemetry.io/collector/config"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filterset"
 )
 
 // Config defines the configuration for the processor.
 type Config struct {
 	config.ProcessorSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
 
-	// List of cumulative metrics to convert to delta.
-	Metrics []string `mapstructure:"metrics"`
-
 	// MaxStaleness is the total time a state entry will live past the time it was last seen. Set to 0 to retain state indefinitely.
 	MaxStaleness time.Duration `mapstructure:"max_staleness"`
+
+	// Deprecated: List of cumulative metrics to convert to delta.
+	// Cannot be used with Include/Exclude.
+	Metrics []string `mapstructure:"metrics"`
+
+	// Include specifies a filter on the metrics that should be converted.
+	// Exclude specifies a filter on the metrics that should not be converted.
+	// If neither `include` nor `exclude` are set, all metrics will be converted.
+	// Cannot be used with deprecated Metrics config option.
+	Include MatchMetrics `mapstructure:"include"`
+	Exclude MatchMetrics `mapstructure:"exclude"`
+}
+
+type MatchMetrics struct {
```

Comment:
Is this based on the config on a different processor? (Asking since the `metricstransform` processor has [a different approach](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/6cb401f1f25d2affcf5a10e737ad1c7e10912206/processor/metricstransformprocessor/config.go#L110) and ideally we want to be as consistent as possible)

Reply:
Ya I copied the Include/Exclude approach from the [hostmetrics receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/6cb401f1f25d2affcf5a10e737ad1c7e10912206/receiver/hostmetricsreceiver/internal/scraper/diskscraper/config.go#L33).  I used this one bc I felt like have both Include and Exclude provided the most flexibility.